RIOT-OS/RIOT
# drivers/ltc4150: (Re-)implemented driver for the LTC4150 coulomb counter

### Contribution description

This PR implements a driver for the LTC4150 coulomb counter. A driver for that device was previously part for RIOT, but was dropped because of quality shortcomings. This driver is written from scratch.


### Testing procedure

The LTC4150 is integrated in the MSB-A2 board, thus using that board will be most convenient to use for testing. However, this chip is also cheaply available as breakout board from various vendors, which all seem to be based on the same open hardware PCB.

Testing:

1. Build, flash & run `tests/driver_ltc4150` on the MSB-A2. I intended the program to be self-explaining - so I will not provide more information here to see if I indeed reached that goal
2. Build, flash & run `example/default`. Try to read the data using the `saul` shell command

### Issues/PRs references

This PR competes with PR https://github.com/RIOT-OS/RIOT/pull/9653 also opened by me. I personally favor this PR strongly over the old approach.
